internal/provider/data_source_available_credentials.go

```diff
@@ -0,0 +1,134 @@
+package provider
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/sigstore/cosign/pkg/cosign/env"
+	"github.com/sigstore/cosign/pkg/providers/filesystem"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &AvailableDataSource{}
+
+func NewAvailableDataSource() datasource.DataSource {
+	return &AvailableDataSource{}
+}
+
+// ExampleDataSource defines the data source implementation.
+type AvailableDataSource struct {
+	popts *ProviderOpts
+}
+
+// ExampleDataSourceModel describes the data source data model.
+type AvailableDataSourceModel struct {
+	Id        types.String `tfsdk:"id"`
+	Available types.Set    `tfsdk:"available"`
+}
+
+func (d *AvailableDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_available_credentials"
+}
+
+func (d *AvailableDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This produces a list of available keyless signing credentials.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "This contains the hash of available keyless signing credentials.",
+				Computed:            true,
+			},
+			"available": schema.SetAttribute{
+				MarkdownDescription: "This contains the names of available keyless signing credentials.",
+				Computed:            true,
+				ElementType:         basetypes.StringType{},
+			},
+		},
+	}
+}
+
+func (d *AvailableDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	popts, ok := req.ProviderData.(*ProviderOpts)
+	if !ok || popts == nil {
+		resp.Diagnostics.AddError("Client Error", "invalid provider data")
+		return
+	}
+	d.popts = popts
+}
+
+// Copied from "github.com/sigstore/cosign/v2/pkg/providers/filesystem"
```

Comment:
Can we reuse https://github.com/chainguard-dev/terraform-provider-cosign/blob/main/internal/provider/oidc.go somehow?

Reply:
I don't know of a good way that doesn't involve a change to that package, e.g., to add `providers.Enabled("github-actions")` or something. I can do that, but I don't know how long it would take, and if there'd be any pushback.